src/templates/generate-ehr-acknowledgement.js

```diff
@@ -47,6 +96,7 @@ const ackMessageTemplate = ({
       </device>
    </communicationFunctionSnd>
    <ControlActEvent classCode="CACT" moodCode="EVN">
+      ${controlActEventReason}
```

Comment:
Would this and line 83 cause an issue in case of positive ack? Because javascript has the nasty habit of displaying null or undefined literally instead of empty string...
```
> let x = null
> `<some-xml> ${x} <some-xml>`
'<some-xml> null <some-xml>'
```

Reply:
Very good point! Well spotted. I've added a couple of lines that'll replace these with empty strings instead.